### PR TITLE
Updates for Dilithium round 3

### PR DESCRIPTION
**Pull request by bhess**
* Updates OIDs for Dilithium round 3 (2, 3, 5, 2_aes, 3_aes, 5_aes)
* Updates security strengths
* Updates `mix_with` according to the security strengths
* `code_point` field is also updated to make the dilithium schemes fit consequently.

Addresses #276 

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated
- [x] Boring interop

### CODE REVIEW

File: oqs-template/generate.yml [hunk 1]
```diff
@@ -326,53 +326,89 @@ sigs:
...
         oqs_meth: 'OQS_SIG_alg_dilithium_2'
         security: 128
-        oid: '1.3.6.1.4.1.2.267.6.4.3'
-        code_point: '0xfe03'
+        oid: '1.3.6.1.4.1.2.267.7.4.4'
+        code_point: '0xfea0'
         enable: true
         mix_with: [{'name': 'p256',
                     'pretty_name': 'ECDSA p256',
                     'oid': '1.3.9999.2.2',
```
Comment (baentsch) [inline oqs-template/generate.yml]: Please bump

File: oqs-template/generate.yml [hunk 2]
```diff
@@ -326,53 +326,89 @@ sigs:
...
-                    'code_point': '0xfe04'},
+                    'code_point': '0xfea1'},
                    {'name': 'rsa3072',
                     'pretty_name': 'RSA3072',
                     'oid': '1.3.9999.2.3',
```
Comment (baentsch) [inline oqs-template/generate.yml]: same here

File: oqs-template/generate.yml [hunk 3]
```diff
@@ -326,53 +326,89 @@ sigs:
...
+        oqs_meth: 'OQS_SIG_alg_dilithium_3_aes'
         security: 192
-        oid: '1.3.6.1.4.1.2.267.6.6.5'
-        code_point: '0xfe09'
+        oid: '1.3.6.1.4.1.2.267.11.6.5'
+        code_point: '0xfeaa'
         enable: true
         mix_with: [{'name': 'p384',
                     'pretty_name': 'ECDSA p384',
                     'oid': '1.3.9999.2.6',
```
Comment (baentsch) [inline oqs-template/generate.yml]: same

File: oqs-template/generate.yml (hunk 1 above)
Comment (bhess) [reply oqs-template/generate.yml:342]: I wasn't completely sure about the implications of `code_point`. Should they rather not be changed?

**Review by baentsch — CHANGES_REQUESTED** [review]
Please bump hybrid OIDs, too. Also, I don't understand why ` crypto/objects/obj_mac.num` gets _shorter_ with _more_ algorithms: Can you explain that?

**Comment by bhess** [comment]
> Please bump hybrid OIDs, too. Also, I don't understand why ` crypto/objects/obj_mac.num` gets _shorter_ with _more_ algorithms: Can you explain that?

Thanks, updating them as well. Also checking the second one..

**Comment by bhess** [comment]
> Also, I don't understand why ` crypto/objects/obj_mac.num` gets _shorter_ with _more_ algorithms: Can you explain that?

Hm, I don't see anything about kems being populated in the jinja-template: https://github.com/open-quantum-safe/openssl/blob/OQS-OpenSSL_1_1_1-stable/oqs-template/crypto/objects/obj_mac.num/assign_ids.fragment

The jinja-snipped might have been accidentally removed sometime.

File: oqs-test/test_tls_full.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import sys
 import os
 
-@pytest.fixture(params=common.signatures)
+@pytest.fixture(params=["p384_dilithium3_aes"])
```
Comment (baentsch) [inline oqs-test/test_tls_full.py]: This doesn't look right.

File: oqs-template/generate.yml (hunk 3 above)
Comment (baentsch) [reply oqs-template/generate.yml]: also looks better with the latest commit

**Comment by baentsch** [comment]
@xvzcf I'd be inclined to accept this PR (after the `test_tls_full` issue has been addressed) even though boring interop fails: Would you agree? Or in other words: Would you have time to look at updating boringssl? Or should @bhess or @baentsch look at that?

**Comment by xvzcf** [comment]
> @xvzcf I'd be inclined to accept this PR (after the `test_tls_full` issue has been addressed) even though boring interop fails: Would you agree? Or in other words: Would you have time to look at updating boringssl? Or should @bhess or @baentsch look at that?

I'll try to update BoringSSL by the end of today and do at least one interop run, but if you or @bhess have spare cycles, I can review both this and the BoringSSL PR.

File: oqs-test/test_tls_full.py (hunk 4 above)
Comment (xvzcf) [reply oqs-test/test_tls_full.py]: Indeed, this shouldn't have to be changed.

**Comment by bhess** [comment]
Just opened a PR for BoringSSL:

https://github.com/open-quantum-safe/boringssl/pull/47

File: oqs-template/generate.yml (hunk 2 above)
Comment (bhess) [reply oqs-template/generate.yml]: done

File: oqs-test/test_tls_full.py (hunk 4 above)
Comment (bhess) [reply oqs-test/test_tls_full.py]: good catch, let's hope that it still works now :)

**Comment by bhess** [comment]
> > @xvzcf I'd be inclined to accept this PR (after the `test_tls_full` issue has been addressed) even though boring interop fails: Would you agree? Or in other words: Would you have time to look at updating boringssl? Or should @bhess or @baentsch look at that?
> 
> I'll try to update BoringSSL by the end of today and do at least one interop run, but if you or @bhess have spare cycles, I can review both this and the BoringSSL PR.

Marking this as ready. The hybrid OIDs are added in OpenSSL (BoringSSL seems to not to have any).
`crypto/objects/obj_mac.num` are fixed with an updated `assign_ids.fragment`. Interop-tests pass locally when pointing to the BoringSSL-PR.

**Comment by xvzcf** [comment]
> The hybrid OIDs are added in OpenSSL (BoringSSL seems to not to have any).

Indeed, BoringSSL currently does not support hybrid signatures.

**Review by baentsch — APPROVED** [review]

File: crypto/objects/objects.txt [hunk 5]
```diff
@@ -1765,14 +1765,20 @@ dstu4145le 2 9 : uacurve9 : DSTU curve 9
...
+1 3 9999 2 7 3 : p384_dilithium3 : p384_dilithium3
+1 3 6 1 4 1 2 267 7 8 7 : dilithium5 : dilithium5
+1 3 9999 2 7 4 : p521_dilithium5 : p521_dilithium5
+1 3 6 1 4 1 2 267 11 4 4 : dilithium2_aes : dilithium2_aes
+1 3 9999 2 11 1 : p256_dilithium2_aes : p256_dilithium2_aes
+1 3 9999 2 11 2 : rsa3072_dilithium2_aes : rsa3072_dilithium2_aes
+1 3 6 1 4 1 2 267 11 6 5 : dilithium3_aes : dilithium3_aes
+1 3 9999 2 11 3 : p384_dilithium3_aes : p384_dilithium3_aes
+1 3 6 1 4 1 2 267 11 8 7 : dilithium5_aes : dilithium5_aes
+1 3 9999 3 11 4 : p521_dilithium5_aes : p521_dilithium5_aes
```
Comment (baentsch) [inline crypto/objects/objects.txt]: This is NOK: Falcon hybrid OID.

File: oqs-template/generate.yml [hunk 6]
```diff
@@ -326,53 +326,89 @@ sigs:
...
+        name: 'dilithium5_aes'
+        pretty_name: 'Dilithium5_AES'
+        oqs_meth: 'OQS_SIG_alg_dilithium_5_aes'
+        security: 256
+        oid: '1.3.6.1.4.1.2.267.11.8.7'
+        code_point: '0xfeac'
+        enable: true
+        mix_with: [{'name': 'p521',
+                    'pretty_name': 'ECDSA p521',
+                    'oid': '1.3.9999.3.11.4',
```
Comment (baentsch) [inline oqs-template/generate.yml]: This is origin of OID error. Please move hybrid dilithium5_aes to hybrid dilithium OID space

**Review by baentsch — CHANGES_REQUESTED** [review]
All other things now OK -- hybrid dil5_aes OID now in wrong OID space, though.

File: oqs-template/generate.yml (hunk 6 above)
Comment (bhess) [reply oqs-template/generate.yml]: Thanks for spotting this! Moved to 1.3.9999.2.11.4

File: crypto/objects/objects.txt (hunk 5 above)
Comment (bhess) [reply crypto/objects/objects.txt]: Updated, resolved in generate.yml

**Review by baentsch — APPROVED** [review]

**Pull request merged**